### PR TITLE
add support urlPattern field

### PR DESCRIPTION
**Pull request by berezkin88**

### CODE REVIEW

File: xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java [hunk 1]
```diff
@@ -116,7 +116,14 @@ private void analyzeRequestUrl(Request.Builder builder, String requestUrl) {
 
     @SuppressWarnings("unchecked")
     private Optional<String> getStubRequestUrl(Map<String, Object> jsonFile) {
-        return Optional.of((String) ((Map<String, Object>) jsonFile.get(REQUEST)).get("url"));
+        Map<String, Object> json = (Map<String, Object>) jsonFile.get(REQUEST);
+        if (json.containsKey("url")) {
+            return Optional.of((String) json.get("url"));
+        } else if (json.containsKey("urlPattern")) {
+            return Optional.of((String) json.get("urlPattern"));
+        } else {
```
Comment (spetrychenko) [inline xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java:124]: 2nd else is not necessary

Comment (berezkin88) [reply xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java:124]: Is it? This shouldn't be the case but what if the Map doesn't have `url` or `urlPattern`, e.g. typo - `urlPatern`?

Comment (berezkin88) [reply xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java:124]: Screenshot 2020-09-15 at 10 13 07

Comment (spetrychenko) [reply xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java:124]: I mean you can omit the last else statement and just return empty optional

Comment (berezkin88) [reply xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java]: I see, adjusted

**Review by spetrychenko — APPROVED** [review]

**Pull request merged**